c_src/h3.c

```diff
@@ -617,6 +723,126 @@ erl_indices_are_neighbors(ErlNifEnv * env, int argc, const ERL_NIF_TERM argv[])
                                                                   : ATOM_FALSE;
 }
 
+static ERL_NIF_TERM
```

Comment:
I don't think this function should be exposed? It's really only useful for the C API to determine how large of a chunk of memory to allocate.

Reply:
I thought this complemented the `max_k_ring_size` function that already existed.

It's useful from the standpoint of "are N indices too many indices to try to polyfill against."

For example:

```erl
1> Polygon = [[{-19.31114335506464,-55.1953125},
  {-19.31114335506464,72.7734375},
  {63.39152174400882,72.7734375},
  {63.39152174400882,-55.1953125},
  {-19.31114335506464,-55.1953125}]].
2> h3:max_polyfill_size(P, 10).
1295404369
```

I might decide that ~1.3 billion H3 indices is too many to return and not call `h3:polyfill/2`.

Somewhat related: there's a segmentation fault if you request a polyfill that's too large ☹️ 

```erl
3> h3:polyfill(P, 10).
Segmentation fault: 11
```

The segmentation fault appears to be happening from within the actual `polyfill` call out to Uber's C library...so, maybe there needs to be a maximum set of possible returned indices?

I'm able to also trigger a segmentation fault with the following input on `k_ring/2`:

```erl
1> h3:k_ring(646974793235180134, 1024 * 1024 * 512).
Segmentation fault: 11
```

Reply:
Fair enough on exposing `max_polyfill_size`. Let's leave it in for now since we can then leave it up to the app developer to determine whether they want to call polyfill or not. 

That also answers the question as to whether we should limit polyfill _internally_ to a max size. I hesitated on that before since every machine will behave differently. 

For the segfaults: Could those be happening when we call alloc (of either kind) and not checking the result? I think there may be a couple of spots where we don't check for a NULL response out of the alloc calls. 

Assuming we do check for NULL results, and given that we expose max_kring_size and max_polyfill_size I'd probably leave it up to the caller to limit how many they're willing to try to retrieve.